File: Python/LinkedList.py
# Create Node
# Create Linked List
# Add nodes to linked list
# print linked list out

class Node:
    def __init__(self,data):
        self.data = data
        self.next = None
class LinkedList:
    def __init__(self):
        self.head = None
    def insert(self,newNode):
        if self.head is None:
            self.head = newNode
            print(newNode.data)
            
        else:
            # head=> john->ben >none
            lastNode = self.head
            while True:
                if lastNode.next is None:
                    break
                lastNode = lastNode.next

            lastNode.next = newNode
    def printList(self):
         
        if self.head is None:
            print('List is empty')
        else:
            currentNode = self.head
            while True:
                if currentNode is None:
                    break
                # print(currentNode.data)
                currentNode = currentNode.next



# Node=> data,next
# firstNode.data => john, firstNode.next =>None
firstNode = Node("john")
linkedList = LinkedList()
linkedList.insert(firstNode)
secondNode = Node("Ben")
linkedList.insert(secondNode)
thirdNode = Node("brew")
# linkedList.insert(thirdNode)
linkedList.printList()

File: Javascript/LinkedList.js
class LinkedList{
    constructor(){
         this.head = null
        this.tail = null
    }
    append(value){
        const newNode = {value,next: null}
        if(this.tail){
            this.tail.next = newNode
        }
        this.tail = newNode 
        if(!this.head) {
            this.head = newNode
        } 
    }
    toArrey(){
        const elements = []
        let curNode = this.head
        while (curNode){
            elements.push(curNode)
            curNode = curNode.next
        }
        return elements
    }
}

const LinkedList1 = new LinkedList()
LinkedList1.append(1)
LinkedList1.append(5)
LinkedList1.append('tyhjkf')
LinkedList1.append(3)
LinkedList1.append('fhsdjkf')
LinkedList1.append('4324')
console.log(LinkedList1.toArrey())